subsys/shell/Kconfig.backends

```diff
@@ -24,6 +24,12 @@ config SHELL_BACKEND_SERIAL
 
 if SHELL_BACKEND_SERIAL
 
+config SHELL_PROMPT_UART
```

Comment:
you could use kconfig template file here, like we do for logging settings. Since this will be the same for every new transport by using template we ensure that option, help will look the same.